web/bfex/components/data_pipeline/tasks/researchid.py

```diff
@@ -0,0 +1,92 @@
+from bfex.components.scraper.scraper_factory import ScraperFactory
+from bfex.components.scraper.scraper_type import ScraperType
+from bfex.models import Faculty, Document
+from bfex.common.utils import URLs, FacultyNames
+from bfex.common.exceptions import WorkflowException
+from bfex.components.data_pipeline.tasks.task import Task
+from bfex.components.key_generation.rake_approach import *
+
+
+class ResearchIdPageScrape(Task):
+    def __init__(self):
+        super().__init__("ResearchId Page Scrape")
+
+    def is_requirement_satisfied(self, data):
+        """ Checks the requirements for a faculty page scraping are satisfied.
+
+        For a ResearchId page scrape, we get the links from ElasticSearch.
+        :param list data: list of all faculty
+        :return: True if valid data, otherwise false
+        """
+        satisfied = True
+
+        for faculty in data:
+
+            if (not isinstance(faculty, tuple) or
+                    not isinstance(faculty[0], str) or
+                    not isinstance(faculty[1], Scrapp)):
+                satisfied = False
+
+        return satisfied
+
+    def run(self, data):
+
+        """Performs a scraping of a faculty members ResearchId page.
+        :param data is a faculty object
+        :return: last faculty member handled
+        """
+        
+        no_text_count = 0
+        for faculty in data:
+            faculty_name = faculty.name
+    
+            search_results = Faculty.search().query('match', name=faculty_name).execute()
+            if len(search_results) > 1:
+                # Shouldn't happen, but could.
+                raise WorkflowException("Professor id is ambiguous during search ... More than 1 result")
+        
+            faculty = search_results[0]
+            if faculty.research_id is not None:
+                
+                scraper = ScraperFactory.create_scraper(faculty.research_id, ScraperType.RESEARCHID)
+                scrapps = scraper.get_scrapps()
+
+                freebies = scrapps[0]
```

Comment:
can we call this free key words instead of freebies? (I will make more sense to someone in the future

Reply:
It's keywords and description. Do you have a name suggestion for it being both?

Reply:
actually it may be easier for the key gen if the key words were separated from the description................

Reply:
just kidding again

plz call it key words and description then